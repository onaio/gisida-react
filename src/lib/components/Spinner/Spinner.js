import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

require('./Spinner.scss');

const mapStateToProps = (state, ownProps) => {
  const mapId = ownProps.mapId || 'map-1';
  const MAP = state[mapId] || { blockLoad: true };
  return {
    mapId,
    MAP,
  }
}

export class Spinner extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div
        className="spinner"
        id={`spinner-${this.props.mapId}`}
        style={{ display: this.props.MAP.showSpinner ? 'block' : 'none'}} />
    );
  }
}

export default connect(mapStateToProps)(Spinner);

Spinner.propTypes = {
  MAP: PropTypes.objectOf(PropTypes.any).isRequired,
  mapId: PropTypes.string.isRequired,
}