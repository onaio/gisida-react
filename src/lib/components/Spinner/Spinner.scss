.spinner {
    width: 50px;
    height: 50px;
    position: fixed;
    right: 0;
    margin-left: auto;
    margin-right: auto;
    top: 45%;
    background-color: #333;
    z-index: 1001;
    border-radius: 100%;
    -webkit-animation: sk-scaleout 1s infinite ease-in-out;
    animation: sk-scaleout 1s infinite ease-in-out;
  }
  
  @-webkit-keyframes sk-scaleout {
    0% { -webkit-transform: scale(0); }
  
    100% {
      -webkit-transform: scale(1);
      opacity: 0;
    }
  }
  
  @keyframes sk-scaleout {
    0% {
      -webkit-transform: scale(0);
      transform: scale(0);
    }
  
    100% {
      -webkit-transform: scale(1);
      transform: scale(1);
      opacity: 0;
    }
  }

  #spinner-map-1 {
    left: 340px;
  }

  #spinner-map-2 {
    left: 1100px;
  }

  