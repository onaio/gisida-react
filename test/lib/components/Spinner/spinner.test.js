import React from 'react';
import Spinner  from '../../../../src/lib/components/Spinner/Spinner.js'
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

const initialState = {
	"map-1": {blockLoad: false, showSpinner: true}
}

const mockStore = configureStore()
const store = mockStore(initialState);

describe('Spinner', () => {

  it('renders without crashing', () => {
    shallow(
      <Spinner 
        store={store}
        mapId='map-1'
        MAP={{}}
      />
    );
  });

  it('Spinner component renders correctly', () => {
    const componentWrapper = mount(
      <Spinner 
        store={store}
        mapId='map-1'
        MAP={{}}
      />
    );
    expect(toJson(componentWrapper)).toMatchSnapshot();
  })
});