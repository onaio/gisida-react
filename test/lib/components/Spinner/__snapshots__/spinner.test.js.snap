// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Spinner component renders correctly 1`] = `
<Connect(Spinner)
  MAP={Object {}}
  mapId="map-1"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Spinner
    MAP={
      Object {
        "blockLoad": false,
        "showSpinner": true,
      }
    }
    dispatch={[Function]}
    mapId="map-1"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="spinner"
      id="spinner-map-1"
      style={
        Object {
          "display": "block",
        }
      }
    />
  </Spinner>
</Connect(Spinner)>
`;
